src/main/java/com/coherentsolutions/training/aqa/java/zhavrid/util/Screen.java

```diff
@@ -0,0 +1,27 @@
+package com.coherentsolutions.training.aqa.java.zhavrid.util;
+
+import org.apache.commons.io.FileUtils;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+
+import static com.coherentsolutions.training.aqa.java.zhavrid.util.WebDriverSingleton.driver;
+
+public class Screen {
+
+    public Screen takeScreen () {
```

Comment:
1) try to test with the page which has lentgh more than one screen - does your logic captures all the page? 
1) let's wrap this logic up to TestNG listener (on each failure), then you won't need to call this method explicitly from the test